File: services/clinical-service/src/domain/events/PatientDischargedEvent.ts
import { DomainEvent } from './EncounterCreatedEvent';

export class PatientDischargedEvent implements DomainEvent {
    readonly eventType = 'patient.discharged';
    readonly occurredAt: Date;
    readonly payload: Record<string, unknown>;

    constructor(encounterId: string, patientId: string, dischargeDate: Date) {
        this.occurredAt = new Date();
        this.payload = {
            encounterId,
            patientId,
            dischargeDate: dischargeDate.toISOString(),
        };
    }
}

File: services/clinical-service/src/main/java/com/medtrust/clinical/domain/event/PatientDischargedEvent.java
package com.medtrust.clinical.domain.event;

import java.time.Instant;
import java.util.Map;

public record PatientDischargedEvent(
        String encounterId,
        String patientId,
        Instant dischargeDate) implements DomainEvent {

    @Override
    public String eventType() {
        return "patient.discharged";
    }

    @Override
    public Instant occurredAt() {
        return Instant.now();
    }

    @Override
    public Map<String, Object> payload() {
        return Map.of(
                "encounterId", encounterId,
                "patientId", patientId,
                "dischargeDate", dischargeDate.toString());
    }
}
